sdk/src/types/block/public_key/mod.rs

```diff
@@ -82,6 +82,8 @@ pub(crate) mod dto {
             }
         }
     }
+
+    impl_serde_typed_enum_dto!(PublicKey, PublicKeyDto, "public key");
```

Comment:
I think this was intended to be used for the variants

Reply:
I see, but it isn't correct to do it this way because if the type field is wrong this just ignores it.

Reply:
Yes not sure if there is a good way to do this with a macro then
Should I implement the conversion manually only for this type and remove the macro?

Reply:
I just used the macro on the individual type DTOs

Reply:
Hm, that is a problem indeed. In that case I think you have to just define the manual deserialization which checks the type field